app/models/user.rb

```diff
@@ -48,6 +48,28 @@ def email_changed?
     !persisted? || (emails.loaded? && emails.each.any?(&:changed))
   end
 
+  # Unset current primary email
+  #
+  # @return [Boolean] True if current primary emails was set as non primary or there is no
+  # primary email, false if failed.
+  def unset_primary_email
+    primary_email = emails.find(&:primary?)
+    return true unless primary_email
+
+    primary_email.update_attributes(primary: false)
+  end
+
+  # Pick first email and set it as primary email.
+  #
+  # @return [Boolean] True if the new email was set as primary, false if failed or next email
```

Comment:
Say that this method would immediately cause the record to have the attribute set in the database.